### ISSUE TITLE
SAT [strict_mode]: `basic_read.expect_records` must be set

### ISSUE THREAD

**Issue by alafanechere** [airbytehq/airbyte#18222]
## Tell us about the problem you're trying to solve
`basic_read.expect_record` is currently optionally used to declare a file with a list of records expected to receive on a `read` command. Making it mandatory for GA connectors is required to validate record schema and limit regression on `read`.

## Describe the solution you’d like
SAT must enforce the presence of the `expect_record` field in the `basic_read` test section for GA connectors.
I'd like these expect_records to be split by stream for maximum control and granularity in a future coverage computation.

**If a bypass_reason is filled SAT should log a warning.**

Example for a partially seeded sandbox account:
```yaml
strict_mode: strict
tests:
  basic_read:
    - config_path: "secrets/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
      expect_records:
        streams:
           - stream_name: stream_a
             expected_records_path: "integration_tests/a_expected_records.txt"
             extra_fields: no
             exact_order: yes
             extra_records: no
           - stream_name: stream_b
             bypass_reason: "We cannot seed stream b"
```

Example for a not seeded sandbox account:

```yaml
strict_mode: strict
tests:
  basic_read:
    - config_path: "secrets/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
      expect_records:
        bypass_reason: "No stream can be seeded on our sandbox account"
```

**If a bypass_reason is filled SAT should log a warning.**

This will require changing the configuration structure of `expect_records` which is not currently stream-specific:
```yaml
  basic_read:
    - config_path: "secrets/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
      expect_records:
        path: "integration_tests/expected_records.txt"
        extra_fields: no
        exact_order: yes
        extra_records: no
```

**Comment by evantahler** [comment]
Grooming Notes:
* To support the previous SAT file format, we will need to use `oneOf`
  * ... unless not that many GA connectors use expected_records, if not that many, we can just migrate a few of them in the PR.

**Comment by alafanechere** [comment]
After further thinking I realized the approach I described above is a bit far fetched, and we can easily piggyback on what was achieved in this [PR](https://github.com/airbytehq/airbyte/pull/18425): 
- Maintainers can already declare the streams they can't seed with the `empty_streams` + `bypass_reason` 
- We want to check that the set of streams contained in the `expect_records` file matches the set of streams declared in the catalog - the empty streams declared. If the check fails it means more test data can be added to the `expect_records` file or some streams were not declared in empty_streams.

This approach does not require any change to the configuration structure and greatly lowers the complexity of this story.
